src/Castle.Core/DynamicProxy/StandardInterceptor.cs

```diff
@@ -19,11 +19,7 @@ namespace Castle.DynamicProxy
 #if FEATURE_SERIALIZATION
 	[Serializable]
 #endif
```

Comment:
**Off-topic for this review:** Hmmm. How is this class serialisable, I guess this was marked serialisable so an invocation could be serialised? An interceptor can't really be serialised because it isn't part of the contract so implementers of the interface wouldn't know to do that, and it made little sense to be marked serialisable and inherit from MBRO.

Definitely need to review all serialisation even though .NET Core 2.0 brings binary serialisation back.

Reply:
AFAIK, invocations needn't be serializable. My guess is `CacheMappingsAtteibute` &rarr; `CacheKey` &rarr; `ProxyGenerationOptions` &rarr; `IInterceptorSelector` & `IInterceptor` & `StandardInterceptor`. There interface types in that chain obviously don't need to be marked `[Serializable]` since that attribute is not inherited, but implementing types should be. Also, it's possible that a interceptor selector stores a field of interceptors, so I suppose `StandardInterceptor` might have been declared serializable so it would be compatible with that scenario.